src/ng/q.js

```diff
@@ -325,14 +364,15 @@ function qFactory(nextTick, exceptionHandler) {
     }
   }
 
-  function scheduleProcessQueue(state) {
+  function scheduleProcessQueue(promise, state) {
     if (state.processScheduled || !state.pending) return;
```

Comment:
Does this implementation understands that we schedule the resolution of promises only when we do not have it already scheduled?